Source/WebCore/platform/mac/ScrollingEffectsController.mm

```diff
@@ -589,32 +589,56 @@ static inline WheelEventStatus toWheelEventStatus(PlatformWheelEventPhase phase,
     return scrollSnapState == ScrollSnapState::Gliding || scrollSnapState == ScrollSnapState::DestinationReached;
 }
 
-void ScrollingEffectsController::scheduleStatelessScrollSnap()
+void ScrollingEffectsController::scheduleDiscreteScrollSnap(const FloatSize& delta)
 {
     stopScrollSnapAnimation();
-    if (m_statelessSnapTransitionTimer) {
-        m_statelessSnapTransitionTimer->stop();
-        m_statelessSnapTransitionTimer = nullptr;
-    }
+    if (m_discreteSnapTransitionTimer) {
+        m_discreteSnapTransitionTimer->stop();
+        m_discreteSnapTransitionTimer = nullptr;
+        m_recentDiscreteWheelDeltas.append(delta);
+        static constexpr auto numberOfDeltasToStoreForDiscreteScrollSnap = 3;
+        if (m_recentDiscreteWheelDeltas.size() > numberOfDeltasToStoreForDiscreteScrollSnap)
+            m_recentDiscreteWheelDeltas.removeFirst();
+    } else
+        m_recentDiscreteWheelDeltas = { delta };
+
     if (!usesScrollSnap())
         return;
 
-    static const Seconds statelessScrollSnapDelay = 750_ms;
-    m_statelessSnapTransitionTimer = m_client.createTimer([this] {
-        statelessSnapTransitionTimerFired();
+    static const Seconds statelessScrollSnapDelay = 100_ms;
```

Comment:
Wouldn't it make sense to rename this variable to `discreteScrollSnapDelay`?

Reply:
Yes — good catch.